api/internal/managers/linux/infra/util.go

```diff
@@ -27,47 +26,43 @@ func (m *infraManager) waitForNode(ctx context.Context, kcli client.Client) erro
 	return nil
 }
 
-func (m *infraManager) kubeClient() (client.Client, error) {
-	if m.kcli != nil {
-		return m.kcli, nil
+// setupClients initializes the kube, metadata, and helm clients if they are not already set.
+// We need to do it after the infra manager is initialized to ensure that the runtime config is available and we already have a cluster setup
+func (m *infraManager) setupClients(kubeConfigPath string, airgapChartsPath string) error {
+	if m.kcli == nil {
+		kcli, err := clients.NewKubeClient(clients.KubeClientOptions{KubeConfigPath: kubeConfigPath})
+		if err != nil {
+			return fmt.Errorf("create kube client: %w", err)
+		}
+		m.kcli = kcli
 	}
-	kcli, err := kubeutils.KubeClient()
-	if err != nil {
-		return nil, fmt.Errorf("create kube client: %w", err)
-	}
-	return kcli, nil
-}
 
-func (m *infraManager) metadataClient() (metadata.Interface, error) {
-	if m.mcli != nil {
-		return m.mcli, nil
-	}
-	mcli, err := kubeutils.MetadataClient()
-	if err != nil {
-		return nil, fmt.Errorf("create metadata client: %w", err)
+	if m.mcli == nil {
+		mcli, err := clients.NewMetadataClient(clients.KubeClientOptions{KubeConfigPath: kubeConfigPath})
+		if err != nil {
+			return fmt.Errorf("create metadata client: %w", err)
+		}
+		m.mcli = mcli
 	}
-	return mcli, nil
-}
 
-func (m *infraManager) helmClient(rc runtimeconfig.RuntimeConfig) (helm.Client, error) {
-	if m.hcli != nil {
-		return m.hcli, nil
+	if m.hcli == nil {
+		airgapPath := ""
+		if m.airgapBundle != "" {
+			airgapPath = airgapChartsPath
+		}
+		hcli, err := helm.NewClient(helm.HelmOptions{
+			KubeConfig: kubeConfigPath,
+			K0sVersion: versions.K0sVersion,
+			AirgapPath: airgapPath,
+			LogFn:      m.logFn("helm"),
+		})
+		if err != nil {
+			return fmt.Errorf("create helm client: %w", err)
+		}
+		m.hcli = hcli
 	}
 
-	airgapChartsPath := ""
-	if m.airgapBundle != "" {
-		airgapChartsPath = rc.EmbeddedClusterChartsSubDir()
-	}
-	hcli, err := helm.NewClient(helm.HelmOptions{
-		KubeConfig: rc.PathToKubeConfig(),
-		K0sVersion: versions.K0sVersion,
-		AirgapPath: airgapChartsPath,
-		LogFn:      m.logFn("helm"),
-	})
-	if err != nil {
-		return nil, fmt.Errorf("create helm client: %w", err)
-	}
-	return hcli, nil
+	return nil
```

Comment:
I had naively thought that I could just move the initialisation of the clients to the manager initialisation like we do for the kubernetes one, only to later realise that ofc we need to wait for the cluster to exist to be able to do that 🙃 nevertheless I do believe we should make use of the struct properties, set them up and then letting the manager make use of those, but would be curious to hear your thoughts.